LayoutTests/media/audioSession/getUserMedia.html

```diff
@@ -0,0 +1,40 @@
+<!doctype html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <script src="../../resources/testharness.js"></script>
+        <script src="../../resources/testharnessreport.js"></script>
+    </head>
+    <body>
+        <script>
+function waitForStateChange()
+{
+    return new Promise((resolve, reject) => {
+        navigator.audioSession.onstatechange = () => resolve(navigator.audioSession.state);
+        setTimeout(() => reject("no audio session state change, current audio session state is " + navigator.audioSession.state), 3000);
+    });
+}
+
+promise_test(async (test) => {
+    assert_equals(navigator.audioSession.type, "auto");
+    assert_equals(navigator.audioSession.state, "inactive");
+
+    let statePromise = waitForStateChange();
+
+    let localStream = await navigator.mediaDevices.getUserMedia({audio:true});
+    localStream.getTracks().forEach(track => track.stop());
+
+    assert_equals(await statePromise, "active");
+
+    navigator.audioSession.type = "playback";
+    await navigator.mediaDevices.getUserMedia({audio:true}).then(assert_unreached, () => { });
+    localStream = await navigator.mediaDevices.getUserMedia({video:true});
+    localStream.getTracks().forEach(track => track.stop());
```

Comment:
Is this to verify that it is possible to create a video-only stream when the session type is `playback`? If so, maybe add a timeout so the test fails rather than just timing out.

In either case, I think it would be helpful to have a comment explaining what is happening.

Reply:
Yes, this is to verify camera access is not tied to audioSession type, I'll add a comment.
I do not think the test will time out if it ever happened, probably the promise would reject (or the camera capture would start but fail, in which case we could beef up the test by checking onended.